ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -58,12 +78,78 @@ func NewRayServiceReconciler(mgr manager.Manager) *RayServiceReconciler {
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.2/pkg/reconcile
-func (r *RayServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = logf.FromContext(ctx)
+func (r *RayServiceReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	_ = r.Log.WithValues("rayservice", request.NamespacedName)
+	log.Info("reconciling RayService", "service NamespacedName", request.NamespacedName)
+
+	// Get serving cluster instance
+	var err error
+	var rayServiceInstance *rayv1alpha1.RayService
+
+	if rayServiceInstance, err = r.getRayServiceInstance(request); err != nil {
+		err = r.updateState(rayServiceInstance, rayv1alpha1.FAIL_TO_GET_RAYSERVICE, err)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
```

Comment:
can you help me understand, does this kick back to the kubebuilder as a terminal state? when will it come back for retries?

Reply:
I imagine that the err would be raised rarely (e.g. when the rayService disappears in the time between lines 81 and 89) 

For me and others who haven't read the KubeBuilder docs :p -- how does the reconciliation workflow work at a high level?

Reply:
If there is err, it will retry. If no error and return just ctrl.Result{}, the reconcile is done and no retry.

Reply:
In this case, `client.IgnoreNotFound(err)` is an err so it will retry right?